runtime/src/main/java/com/flipkart/flux/controller/WorkFlowExecutionController.java

```diff
@@ -100,36 +118,39 @@ public WorkFlowExecutionController(EventsDAO eventsDAO, StateMachinesDAO stateMa
 
     /**
      * Retrieves the states which are dependant on this event and starts the execution of eligible states (whose all dependencies are met).
+     *
      * @param eventData
      * @param stateMachineInstanceId
      * @param correlationId
      */
-    public Set<State> postEvent(EventData eventData, Long stateMachineInstanceId, String correlationId) {
+    public Set<State> postEvent(EventData eventData, String stateMachineInstanceId, String correlationId) {
         StateMachine stateMachine = null;
         if (stateMachineInstanceId != null) {
             stateMachine = retrieveStateMachine(stateMachineInstanceId);
-        } else if(correlationId != null) {
+        } else if (correlationId != null) {
             stateMachine = retrieveStateMachineByCorrelationId(correlationId);
```

Comment:
all references to retrieveStateMachineByCorrelationId() should be replaced with retrieveStateMachine()